### ISSUE TITLE
HTTP 307 and HTTP 308 Redirects are not handled

### ISSUE THREAD

**Issue by miranda-zhang** [protegeproject/protege#849]
My current configuration used redirect code 308 on
https://w3id.org/cocoon/v1.0
It opens fine in the browser, but if I try “Open from URL”, it throws a
`sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target`

If I try to open with the original link to file, it works fine in protoge.
https://miranda-zhang.github.io/cloud-computing-schema/v1.0/ontology.ttl

My redirect rules are listed here
https://github.com/perma-id/w3id.org/blob/master/cocoon/.htaccess

There is an issue filed with w3id.org, but I also suspect there maybe an issue with protoge too.
https://github.com/perma-id/w3id.org/issues/1063

**Comment by balhoff** [comment]
I have been seeing the same error when trying to open `http://purl.obolibrary.org/obo/mondo.owl`, which is hosted at osf.io.

**Comment by matthewhorridge** [comment]
This is working in the 5.5.0-beta-SNAPSHOT. MONDO loads fine.  

The redirect for coocon lands at a website, so the parse fails (but the redirect works).

**Comment by miranda-zhang** [comment]
> The redirect for coocon lands at a website, so the parse fails (but the redirect works).

I have tested with `curl -LH "Accept: application/rdf+xml" https://w3id.org/cocoon/v1.0`
It returns the XML document.
What request header is protoge using?

**Comment by matthewhorridge** [comment]
Hi @miranda-zhang,

> What request header is protoge using? Or is there anything wrong with my content negotiation?

It uses request headers set by the OWL API, and ```application/rdf+xml``` is included in these.  I'll do some more investigating and get back to you

**Comment by matthewhorridge** [comment]
@miranda-zhang 

I've just done some digging and I've found out that HTTP 308 redirects are in fact *not* followed.  The OWL API that Protege uses follows HTTP 301 Moved Permanently, HTTP 302 Found and HTTP 303 See Other redirects.

As a workaround, are you able to change your redirect to a HTTP 301?

It seems like HTTP 308 (and HTTP 307 for that matter) redirects could/should be supported in the OWL API.  I'll file a request for this.


p.s. This is the accept header for a request

```
application/ld+json; q=1.0, application/x-trig; q=1.0, application/owl+xml; q=1.0, application/rdf+xml; q=1.0, application/x-binary-rdf; q=1.0, text/owl-manchester; q=1.0, text/turtle; q=1.0, application/trix; q=1.0, application/xhtml+xml; q=1.0, text/owl-functional; q=1.0, application/rdf+json; q=1.0, text/n3; q=1.0, text/plain; q=1.0, text/x-nquads; q=1.0, application/xml; q=0.5, application/x-turtle; q=0.5, application/html; q=0.5, text/xml; q=0.5, text/rdf+n3; q=0.5, text/html; q=0.3, */*; q=0.09
```

**Comment by matthewhorridge** [comment]
I've filed an issue for this on the OWL API tracker: https://github.com/owlcs/owlapi/issues/821

**Comment by matthewhorridge** [comment]
Pushing this to 5.6.0 milestone because the beta-9 milestone seems stable and I'm reluctant to update the OWL API at this juncture.

**Comment by matthewhorridge** [comment]
Closed with the upgrade to the OWL API 4.5.11.  Note @miranda-zhang that the RDF/XML file is not found (404 error) but this nothing to do with Protege.
